### PR TITLE
fix: retain receipt when editing, remove code for advance wallet id, set unspecified category id

### PR DESCRIPTION
**Pull request by arjunaj5**
## Clickup
Please add link here

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 1]
```diff
@@ -4082,17 +4083,6 @@ export class AddEditExpensePage implements OnInit {
...
-              }),
-              switchMap((txn) => {
-                if (txn.id && txn.advance_wallet_id !== etxn.tx.advance_wallet_id) {
-                  const expense = {
-                    id: txn.id,
-                    advance_wallet_id: etxn.tx.advance_wallet_id,
-                  };
-                  return this.expensesService.post(expense).pipe(map(() => txn));
-                } else {
-                  return of(txn);
```
Comment (arjunaj5) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:4094]: `advance_wallet_id` will be already set in the normal POST flow. We don't need this additional call

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 2]
```diff
@@ -4036,10 +4040,7 @@ export class AddEditExpensePage implements OnInit {
               report: Report;
             };
 
-            // NOTE: This double call is done as certain fields will not be present in return of upsert call. policy_amount in this case.
             return this.transactionService.upsert(etxn.tx as Transaction).pipe(
-              switchMap((txn) => this.expensesService.getExpenseById(txn.id)),
-              map((expense) => this.transactionService.transformExpense(expense).tx),
```
Comment (arjunaj5) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:4042]: we are not using the result of this double call for checking `policy_amount` and other fields

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 3]
```diff
@@ -3545,7 +3548,8 @@ export class AddEditExpensePage implements OnInit {
           amount = etxn.tx.user_amount;
         }
 
-        //TODO: Add depenedent fields to custom_properties array once APIs are available
+        const category_id = this.getOrgCategoryID() || unspecifiedCategory.id;
```
Comment (arjunaj5) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:3551]: setting unspecified category as the default category.

File: src/app/core/services/transaction.service.ts [hunk 4]
```diff
@@ -205,7 +205,8 @@ export class TransactionService {
           }
         } else {
           const fileIds = fileObjs.map((fileObj) => fileObj.id);
-          return this.upsert(txn, fileIds);
+          txn.file_ids = fileIds;
```
Comment (arjunaj5) [inline src/app/core/services/transaction.service.ts:208]: only during `createTxnWithFiles` do we have to set the `file_ids` in the add/edit flow. In other scenarios of the add/edit, the file_ids will be already present.

**Review by mvaishnavi — APPROVED** [review]

**Pull request merged**